src/languages/scala.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+  cascadingMatcher,
+} from '../util/nodeMatchers';
+import { NodeMatcherAlternative, ScopeType } from '../typings/Types';
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  // treating classes = classlike
+  class: ['class_definition', 'object_definition', 'trait_definition'],
+  className: ['class_definition[name]', 'object_definition[name]', 'trait_definition[name]'],
+
+  ifStatement: 'if_expression',
+
+  string: ['interpolated_string_expression', 'string'],
+  comment: 'comment',
+
+  // lists basic definition is just a function call to a constructor, eg List(1,2,3,4)
+  // MISSING: fancy list style: val foo = 1 :: (2 :: (3 :: Nil)) // List(1,2,3)
+  list: 'call_expression',
+  map: 'call_expression',
```

Comment:
Hmm these seem a bit generic.  Maybe it would be better to check the name of the function that is being called to ensure it is `List` or `Map`?  You'll have to write a custom node finder, but you just check type of the parent node, then access the function call node and check `node.text`.  Lmk if you want to pair on this one.

If you do get it working, it's worth adding a test case where you have a list, one of whose elements is a function call, to ensure that when you say "clear list" within that function call you get the outer list rather than just the call

Reply:
For now I'm going to drop these, as users can use "call" just as effectively